app/src/services/projects.ts

```diff
@@ -368,86 +368,122 @@ async function toastedProjectOperation(label: string, fn?: () => Promise<void>)
 }
 
 export async function createProject(project: SmartContractProjectCreate) {
-    await toastedProjectOperation('Create project', async () => {
-        const chainId = getProjectRegistryChainId()
+    await toastedProjectOperation('Create project', () => {
+        return new Promise(async (resolve, reject) => {
+            const chainId = getProjectRegistryChainId()
 
-        const {
-            id,
-            paymentDetails,
-            streams,
-            minimumSubscriptionInSeconds,
-            isPublicPurchasable,
-            metadata,
-        } = project
-
-        const from = await getDefaultWeb3Account()
-
-        await networkPreflight(chainId)
-
-        await getProjectRegistryContract(chainId, getWeb3())
-            .methods.createProject(
+            const {
                 id,
-                getDomainIds(paymentDetails),
-                getPaymentDetails(paymentDetails),
+                paymentDetails,
                 streams,
                 minimumSubscriptionInSeconds,
                 isPublicPurchasable,
                 metadata,
-            )
-            .send({
-                from,
-                maxPriorityFeePerGas: null,
-                maxFeePerGas: null,
-            })
+            } = project
+
+            const from = await getDefaultWeb3Account()
+
+            await networkPreflight(chainId)
+
+            const tx = getProjectRegistryContract(chainId, getWeb3())
+                .methods.createProject(
+                    id,
+                    getDomainIds(paymentDetails),
+                    getPaymentDetails(paymentDetails),
+                    streams,
+                    minimumSubscriptionInSeconds,
+                    isPublicPurchasable,
+                    metadata,
+                )
+                .send({
+                    from,
+                    maxPriorityFeePerGas: null,
+                    maxFeePerGas: null,
+                })
+
+            tx
```

Comment:
Why not `.send(…).on(…).on(…)`?

It's possible and a bit cleaner if you don't return `tx` (not returning it is what you want).

[Working chaining exemple here.](https://github.com/streamr-dev/core-frontend/blob/thehub/app/src/shared/stores/purchases.ts#L493)